internal/gen/stringutils/stringutils.go

```diff
@@ -1,3 +1,17 @@
+// Copyright 2018 Twitch Interactive, Inc.  All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may not
+// use this file except in compliance with the License. A copy of the License is
+// located at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// or in the "license" file accompanying this file. This file is distributed on
+// an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+//
+// This file contains some code from https://github.com/golang/protobuf:
```

Comment:
Is this an appropriate way to handle the fact that this file is a mix of github.com/golang/protobuf's code and our own?

Reply:
Yes, that's appropriate. 

We should also include that higher up in the licensing (license/notice etc). I'll make a pull request.